### ISSUE TITLE
[Semaphore] Check if key is expired

### ISSUE THREAD

**Issue by jderusse** [symfony/symfony#38583]
In lock component, I setup a mechanism to check if key is expired while trying to acquiring it (ie, because of latency in network, it took more time than expected, and the key expired during the process).

In most of case it never happens, but the probability increases when:
- the TTL is low
- the Store needs several operation to acquire the lock (ie. CombinedStore)
- the backend is frozen (SGDB locks table because of maintenance/dump, in a PdoStore, ...)

This mechanism prevent thinking the lock is acquired, while it's not.

This has not been implemented in Semaphore. Given the two component should provide the same reliability I wonder:
- is it overkill and should be removed from Lock component
- is it needed and should be added to Semaphore component

/cc @lyrixx

**Comment by lyrixx** [comment]
Ohhh I understand better the lock component and this feature! I did not understand why there was this piece of code in the Lock component in the first place, that's why I did not "ported" it to the semaphore component.

So to answer your question: it looks like this feature is required in the **lock component. So I would keep it.**

But in the Semaphore component, we have only one backend (redis) for now. And redis is reputed to be super fast with a low TTL. So all cases where we would need to check that could not exist in the semaphore component. 

So I would **do nothing for now in the Semaphore component**. And If the uses cases become valid, we will implement this check.

But if one come with a PR to address that, I'll not reject it (Note: A nice comment in the could would be required to be merged :p)
